### ISSUE TITLE
ctx.is_training and use_local_stat question about models.BatchNorm function

### ISSUE THREAD

**Issue by hhxjzyr** [ppwwyyxx/tensorpack#554]
In the [Inference tutorial](http://tensorpack.readthedocs.io/en/latest/tutorial/inference.html#inference)： 

**Inference during training** and **Inference after training** is strictly distinguished.

For example, I would like to use an `InferenceRunner` as a callback to do **Inference during training** 

```
    return TrainConfig(
        dataflow=dataset_train,
        callbacks=[
            ModelSaver(),
            InferenceRunner(dataset_test,
                [ScalarStats('cost'), ClassificationError()]),
            ScheduledHyperParamSetter('learning_rate',
                                      [(1, 0.1), (args.drop_1, 0.01), (args.drop_2, 0.001)])
        ],
        model=Model(depth=args.depth),
        steps_per_epoch=steps_per_epoch,
        max_epoch=args.max_epoch,
    )
```

However, in `models.BatchNorm` function, if I  do **Inference during training**  as above code, in this **Inference during training**'s  *inference* stage:
1.  is `ctx.is_training` True or False?
2.  is `use_local_stat`  defaults to True or False?

Should we still consider **Inference during training** still as  **training** and `ctx.is_training==False`  `use_local_stat==False`

**Comment by ppwwyyxx** [comment]
they are False

**Comment by hhxjzyr** [comment]
Thank you very much!
